### PR TITLE
Increase Crawler Keywords

### PR DESCRIPTION
**Pull request by Roseart**
- Change texts and keywords to lowercase for comparisons
- Expand keywords to fires that happened during the past month

### CODE REVIEW

File: backend/data_preparation/crawler/twitter_filter_api_crawler.py [hunk 1]
```diff
@@ -60,18 +61,21 @@ def crawl(self, keywords: List, batch_number: int = 100) -> List[int]:
                     self.reset_wait_time()
 
                     has_keywords = set(self.keywords) & self._tokenize_tweet_text(tweet)
```
Comment (Yicong-Huang) [inline backend/data_preparation/crawler/twitter_filter_api_crawler.py:63]: this line did not use keywords in lower case. DO change the self.keywords to be contain lower case only

File: backend/data_preparation/crawler/twitter_filter_api_crawler.py [hunk 2]
```diff
@@ -60,18 +61,21 @@ def crawl(self, keywords: List, batch_number: int = 100) -> List[int]:
...
+                    # if the tweet is sent within US and contains keywords, add its id to cache and data (for return)
                     elif tweet.get('place') and tweet['place']['country_code'] == "US" and has_keywords:
                         self._add_to_batch(tweet['id'])
                     else:
                         continue
 
-                    count = len(self.data)
-                    if count % (batch_number // 10) == 0:
-                        logger.info(f"Crawled ID count in this batch: {count}")
+                    # print Crawling info once every 10 tweets are collected
```
Comment (Yicong-Huang) [inline backend/data_preparation/crawler/twitter_filter_api_crawler.py]: 10 is depending on the batch number. not a static number.
please update the comment

File: backend/data_preparation/crawler/twitter_filter_api_crawler.py [hunk 3]
```diff
@@ -60,18 +61,21 @@ def crawl(self, keywords: List, batch_number: int = 100) -> List[int]:
...
+                    if len(self.data) > count:
```
Comment (Yicong-Huang) [inline backend/data_preparation/crawler/twitter_filter_api_crawler.py:77]: when count is already 100, while loop restarts, this if statement will always be false?

**Review by Yicong-Huang — DISMISSED** [review]
there are some logical errors. please take care of them

**Comment by Yicong-Huang** [comment]
This PR contains 3 duplicate commits from master. Please make sure you are branching from the current master.

File: backend/task/text_from_twitter_runnable.py [hunk 4]
```diff
@@ -50,8 +51,11 @@ def run(self, keywords: list = None, batch_num: int = 100, using_filter_api: boo
             self.crawler = TweetFilterAPICrawler()
 
         while True:
+            fire_names = {fire_name.lower() for fire_name, in Connection.sql_execute(
+                f"select name from fire_merged where start_time > now() - interval '1 month'")}
+            keywords_with_fire_names = fire_names | (set(keywords))
```
Comment (Yicong-Huang) [inline backend/task/text_from_twitter_runnable.py:57]: add a log here to output all the keywords that are explicitly being used

File: backend/data_preparation/crawler/twitter_filter_api_crawler.py (hunk 3 above)
Comment (Roseart) [reply backend/data_preparation/crawler/twitter_filter_api_crawler.py:77]: The function is called again once the batch number of tweets are crawled. Then there's no need to reset the count number.

**Review by Yicong-Huang — APPROVED** [review]
LGTM

**Pull request merged**